### PR TITLE
Unify OpenMP control

### PR DESCRIPTION
**Pull request by rijobro**

### CODE REVIEW

File: SuperBuild/External_Gadgetron.cmake [hunk 1]
```diff
@@ -78,6 +78,12 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
   option(${proj}_USE_CUDA "Enable ${proj} CUDA (if cuda libraries are present)" ${CUDA_FOUND})
   mark_as_advanced(${proj}_USE_CUDA)
 
+  if (NOT DISABLE_OpenMP)
+    option(${proj}_ENABLE_OPENMP "Build ${proj} with OpenMP acceleration" ON)
+  else()
+    option(${proj}_ENABLE_OPENMP "Build ${proj} with OpenMP acceleration" OFF)
```
Comment (KrisThielemans) [inline SuperBuild/External_Gadgetron.cmake]: if it's disabled, then we shouldn't provide the option I think, i.e. just use `set(${proj}_ENABLE_OPENMP OFF)`

Comment (rijobro) [reply SuperBuild/External_Gadgetron.cmake]: done.

**Comment by rijobro** [comment]
@KrisThielemans I wanted to get this one merged, any idea why STIR is failing on linux? 

I put a `find_package(openmp...` at the SB level. If present, openMP support is enabled in STIR. But this is the same behaviour as before, where openMP support in STIR was enabled by default for linux. 

Given the negligible changes to `External_STIR.cmake`, I don't understand why STIR is now failing.

**Comment by KrisThielemans** [comment]
@rijobro could you paste the relevant error, or maybe it isn't clear...

**Comment by rijobro** [comment]
Error happens during the cmake configure step for STIR:
```
CMake Error at /home/travis/.local/bin/cmake-3.13.4-Linux-x86_64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenMP_C (missing: OpenMP_gomp_LIBRARY
  OpenMP_pthread_LIBRARY) (found version "4.5")
Call Stack (most recent call first):
  /home/travis/.local/bin/cmake-3.13.4-Linux-x86_64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /home/travis/.local/bin/cmake-3.13.4-Linux-x86_64/share/cmake-3.13/Modules/FindOpenMP.cmake:473 (find_package_handle_standard_args)
  src/CMakeLists.txt:174 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/travis/build/CCPPETMR/SIRF-SuperBuild/builds/STIR/build/CMakeFiles/CMakeOutput.log".
CMakeFiles/STIR.dir/build.make:106: recipe for target 'builds/STIR/stamp/STIR-configure' failed
make[2]: *** [builds/STIR/stamp/STIR-configure] Error 1
CMakeFiles/Makefile2:493: recipe for target 'CMakeFiles/STIR.dir/all' failed
make[1]: *** [CMakeFiles/STIR.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
Note: checking out 'b6191eaaa72ccca6c6a5fe4c0fa3319694f512ab'.
```

**Comment by KrisThielemans** [comment]
I'm not sure about this. I guess the question is which of all these variables we need to cache. 
```
  mark_as_superbuild(ALL_PROJECTS VARS 
      OpenMP_CXX_FLAGS:STRING OpenMP_CXX_LIB_NAMES:STRING OpenMP_C_FLAGS:STRING 
      OpenMP_C_LIB_NAMES:STRING OpenMP_CXX_FLAGS:STRING
      OpenMP_CXX_LIB_NAMES:STRING
      OpenMP_libomp_LIBRARY:FILEPATH OpenMP_omp_LIBRARY:FILEPATH)
```
I wonder if some of them are not set and then `FindOpenMP.cmake` gets itself in a muddle.

in particular the ` OpenMP_libomp_LIBRARY` and `OpenMP_omp_LIBRARY` are not documented on https://cmake.org/cmake/help/latest/module/FindOpenMP.html. Maybe we should only pass them on if they're set.

I note that the doc says

> OpenMP_<lang>_INCLUDE_DIR cache variable. Note that this variable is an _input_ control to the module. Project code should use the OpenMP_<lang>_INCLUDE_DIRS _output_ variable if it needs to know what include directories are needed.

which might imply we'd need to set `OpenMP_<lang>_INCLUDE_DIR` from `OpenMP_<lang>_INCLUDE_DIRS`....

**Pull request merged**